pkg/resource/tls_secret.go

```diff
@@ -156,14 +161,20 @@ func (s *TLSSecret) IsRotationRequired(duration time.Duration, cronStr string) (
 		return true, "Failed to verify expiry date due to invalid cron, rotating certificate"
 	}
 
-	nextRun := cronSchedule.Next(time.Now())
+	nextRun := cronSchedule.Next(now)
+	nextToNextRun := cronSchedule.Next(nextRun)
 
-	if expiryTime.Before(nextRun) {
+	// if cert is expiring before next run or within (next run + 1 hour)
+	if expiryTime.Sub(nextRun) < 1*time.Hour {
 		return true, "Certificate about to expire, rotating certificate"
 	}
 
-	return false, ""
+	// if cert is expiring before next-to-next run or within (next-to-next run + 1 hour)
+	if expiryTime.Sub(nextToNextRun) < 1*time.Hour {
+		return true, "Certificate about to expire, rotating certificate"
+	}
```

Comment:
@prafull01 your comment does make sense. I don't exactly remember the scenario since it was more than a month ago (should have documented the issue properly). I will remove this and update the unit test in next patch set.

Reply:
Apologies for addressing your comment late. I have just pushed a new patchset.